src/App.tsx

```diff
@@ -1,48 +1,56 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import classNames from 'classnames';
+import {
+  Routes, Route, NavLink, Navigate,
+} from 'react-router-dom';
 
-// const tabs = [
-//   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
-//   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
-//   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
-// ];
+import TabsPage from './components/Tabs';
+import { HomePage } from './components/HomePage';
+import { NotFoundPage } from './components/NotFoundPage';
 
-const App = () => (
-  <>
-    {/* Also requires <html class="has-navbar-fixed-top"> */}
-    <nav className="navbar is-fixed-top has-background-light" data-cy="nav">
-      <div className="navbar-menu">
-        <div className="navbar-start">
-          <a href="/" className="navbar-item isActive">Home</a>
-          <a href="/tabs" className="navbar-item isActive">Tabs</a>
-        </div>
-      </div>
-    </nav>
+const tabs = [
+  { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
+  { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
+  { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
+];
+
+const App = () => {
+  const navLinkClassNames = ({ isActive }: { isActive: boolean }) => {
+    return classNames('navbar-item', { 'is-active': isActive });
+  };
 
-    <div className="section">
-      <h1 className="title">Home page</h1>
-      <h1 className="title">Tabs page</h1>
-      <h1 className="title">Page not found</h1>
+  return (
+    <>
+      <nav className="navbar is-fixed-top has-background-light" data-cy="nav">
+        <div className="navbar-menu">
+          <div className="navbar-start">
+            <NavLink
+              className={navLinkClassNames}
+              to="/"
+            >
+              Home page
+            </NavLink>
 
-      <div className="tabs is-boxed">
-        <ul>
-          <li data-cy="tab" className="is-active">
-            <a href="#/">Tab 1</a>
-          </li>
-          <li data-cy="tab">
-            <a href="#/">Tab 2</a>
-          </li>
-          <li data-cy="tab">
-            <a href="#/">Tab 3</a>
-          </li>
-        </ul>
-      </div>
+            <NavLink
+              className={navLinkClassNames}
+              to="/tabs"
+            >
+              Tabs Page
+            </NavLink>
+          </div>
+        </div>
+      </nav>
 
-      <div className="block" data-cy="tab-content">
-        Please select a tab
-      </div>
-    </div>
-  </>
-);
+      <Routes>
+        <Route path="/" element={(<HomePage />)} />
+        <Route path="/home" element={<Navigate to="/" />} />
+        <Route path="tabs/:tabId" element={<TabsPage tabs={tabs} />} />
+        <Route path="/tabs" element={<TabsPage tabs={tabs} />} />
```

Comment:
use nested routes, for more info, react [this](https://reactrouter.com/docs/en/v6/getting-started/concepts#index-routes) out